verification.py

```diff
@@ -758,6 +759,271 @@ def test_sweep_ex1(self):
+            print(file=output)
+            print("//", file=output)
+            if msms_params["theta"] > 0:
+                s = tree_sequence.get_num_mutations()
+                print("segsites:", s, file=output)
+
+                if s != 0:
+                    print("positions: ", end="", file=output)
+                    positions = [
+                        mutation.position / msms_params["num_sites"]
+                        for mutation in tree_sequence.mutations()
+                    ]
+                    positions.sort()
+                    for position in positions:
+                        print(
+                            "{0:.{1}f}".format(position, 8), end=" ", file=output,
+                        )
+                    print(file=output)
+                    for h in tree_sequence.haplotypes():
+                        print(h, file=output)
+
+                else:
+                    print(file=output)
+        output.close()
+
+        # pipe ms format output to sample_stats
+        p1 = subprocess.Popen(["cat", temp_file], stdout=subprocess.PIPE)
+        p2 = subprocess.Popen(
+            ["./data/sample_stats"], stdin=p1.stdout, stdout=subprocess.PIPE
+        )
+        p1.stdout.close()
+        output = p2.communicate()[0]
+        p1.wait()
+
+        # read into pandas frame and return it
+        with tempfile.TemporaryFile() as f:
+            f.write(output)
+            f.seek(0)
+            df = pd.read_csv(f, sep="\t")
+        return df
+
+    def _run_msms_sample_stats(self, cmd):
+        return self._run_sample_stats(_msms_executable + cmd.split(" "))
+
+    def _convert_to_discoal_cmd(self, msms_cmd):
+        """ called by _run_discoal_sample_stats to convert msms cmdline args
+        to discoal cmdline args
+        """
+        params = self._msms_str_to_parameters(msms_cmd)
+        return "%d %d %d -t %f -r %f -ws %f -a %f -x %f" % (
+            params["nsam"],
+            params["nrep"],
+            params["num_sites"],
+            params["theta"],
+            params["rho"],
+            params["end_time_lst"][0],
+            params["alpha"],
+            params["sel_pos"],
+        )
+
+    def _run_discoal_sample_stats(self, msms_cmd):
+        discoal_cmd = self._convert_to_discoal_cmd(msms_cmd)
+        return self._run_sample_stats(_discoal_executable + discoal_cmd.split(" "))
+
+    def _run_msms_vs_msp(self, cmd):
+        df_msp = self._run_msp_sample_stats(cmd)
+        df_msms = self._run_msms_sample_stats(cmd)
+        self._plot_stats("msp_msms", df_msp, df_msms, "msp", "msms")
+
+    def _run_msms_vs_discoal(self, cmd):
+        df_discoal = self._run_discoal_sample_stats(cmd)
+        df_msms = self._run_msms_sample_stats(cmd)
+        self._plot_stats("discoal_msms", df_discoal, df_msms, "discoal", "msms")
+
+    def test_Neutral_msms_vs_msp(self):
+        self._run_msms_vs_msp("100 300 -t 200 -r 200 500000 -N 10000")
+
+    def test_Selective_msms_vs_msp(self):
```

Comment:
I've created an issue to track this: #1111, can you mention this in the comment please?

Reply:
@jeromekelleher I am glad you caught this. This happens randomly but pretty frequently. In my hand, it could happen in any of the tests involved with the msprime sweep model. I can try to use smaller numbers in the cmd line to see if this can go away for now.